kie-based-services/src/main/java/io/elimu/serviceapi/config/ConfigAPIUtil.java

```diff
@@ -84,16 +86,24 @@ public Map<String, Object> getConfig(ServiceRequest request, String environment,
 						} else {
 							LOG.warn("Invalid value type for JSON property: " + node.getNodeType().name());
 						}
+						}
 					}
+					else {
+						throw new TimeoutException("Timeout occurred when fetching configuration parameters");
+					}
+					CACHE.put(url, new CachedResult(retval));
+				} catch (Exception e) {
```

Comment:
Suggestion, adding a catch block here like this:
`catch (TimeoutException e) { throw e; }`
to not nest the exception unnecessarily